robots/EDUCATOR/gyro.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+"""
+Move robot in a square path using the Gyro sensor.
+
+This script is a simple demonstration of turning using the Gyro sensor.
+"""
+
+from ev3dev.motor import MoveSteering
+from ev3dev.sensor.lego import GyroSensor
+
+
+class GyroSensorEnhanced(GyroSensor):
+    """Enhanced Gyro."""
+
+    def __init__(self):
+        """Create Gyro object if connected."""
+        super().__init__()
+        assert self.connected, 'Connect a Gyro to a sensor port'
+
+    def wait_until_changed_by(self, delta):
+        """Wait until angle has changed by specified amount."""
+        start_angle = self.value()
+        while abs(start_angle - self.value()) < delta:
```

Comment:
Hmmmm... busy-waiting is a bit scary as the EV3 has so little processing power this could entirely lock up the brick. Can we add a short sleep?

Reply:
I'd be game for putting this to the `GyroSensor` class instead...other gyro programs may find this handy

Reply:
Thinking outloud...would be cool if long term we can make all of the sensor files work with poll()

Reply:
Okay if I raise a pull request to add wait_until_changed_by() to the GyroSensor class ? I’ll include a short sleep.

Reply:
sounds good